controller/integration/test/controlflow/controlflow_test.go

```diff
@@ -83,158 +83,155 @@ func Test_Finalizer_Integration_Controlflow(t *testing.T) {
 
 			return nil
 		}
-		b := backoff.NewExponential(2*time.Minute, 10*time.Second)
+		b := backoff.NewExponential(2*time.Second, 10*time.Second)
 
 		err = backoff.Retry(o, b)
 		if err != nil {
 			t.Fatal("expected", nil, "got", err)
 		}
 	}
 
-	// We use backoff with the absolute maximum amount:
-	// 10 second ResyncPeriod + 2 second RateWait + 8 second for safety.
-	// The controller should now add the finalizer and EnsureCreated should be hit
-	// once immediatly.
+	// Verify we reconcile creation of that object. We should have also
+	// 2 ResyncPeriods in 30 seconds so we check if there were more than
+	// 2 create events.
 	//
-	// 		EnsureCreated: 1, EnsureDeleted: 0
+	// 		EnsureCreated: >2, EnsureDeleted: =0
 	//
-	// The controller should reconcile once in this period.
-	//
-	// 		EnsureCreated: 2, EnsureDeleted: 0
-	//
-	operation := func() error {
-		if tr.CreateCount() != 2 {
```

Comment:
We can't really rely on equality on those tests other than making sure that something is zero (i.e. situation we expect thing to not happen at all).